NuRadioReco/modules/trigger/rnog_surface_trigger.py

```diff
@@ -0,0 +1,214 @@
+        triggered_channels: array of ints or None
+            channels ids that are triggered on, if None trigger will run on all channels
+        coinc_window: float
+            time window in which number_coincidences channels need to trigger
+        temp: float
+            temperature of the board with diode in Kelvin
+        Vbias: float
+            applied bias voltage to the diode board in Volt
+        trigger_name: string
+            a unique name of this particular trigger
+        """
+
+        t = time.time()  # absolute time of system
+
+        sampling_rate = station.get_channel(det.get_channel_ids(station.get_id())[0]).get_sampling_rate()
+        dt = 1. / sampling_rate
+
+        triggered_bins_channels = []
+        channels_that_passed_trigger = []
+
+        if triggered_channels is None:  # caveat: all channels start at the same time
+            for channel in station.iter_channels():
+                channel_trace_start_time = channel.get_trace_start_time()
+                break
+        else:
+            channel_trace_start_time = station.get_channel(triggered_channels[0]).get_trace_start_time()
+
+        for channel in station.iter_channels():
+            channel_id = channel.get_id()
+            logger.debug(f'channel id {channel_id}')
+            if triggered_channels is not None and channel_id not in triggered_channels:
+                logger.debug("skipping channel {}".format(channel_id))
+                continue
+            if channel.get_trace_start_time() != channel_trace_start_time:
+                logger.warning('Channel has a trace_start_time that differs from '
+                               '        the other channels. The trigger simulator may not work properly')
+
+            frequencies = channel.get_frequencies()
+            logger.debug(f'trace before trigger {np.abs(np.max(channel.get_trace()))}')
+            #here we apply a bandpass filter which is in the signal chain
+            f = np.zeros_like(frequencies, dtype=complex)
+            mask = frequencies > 0
+            b, a = scipy.signal.cheby1(3, 5, [80 * units.MHz, 180 * units.MHz], 'bandpass', analog=True)  # Numerator (b) and denominator (a) polynomials of the IIR filter
+            #b, a = scipy.signal.butter(10, [80 * units.MHz, 180 * units.MHz], 'bandpass', analog=True)  # Numerator (b) and denominator (a) polynomials of the IIR filter
+            w, h = scipy.signal.freqs(b, a, frequencies[mask])  # w :The angular frequencies at which h was computed. h :The frequency response.
+            f[mask] = h
+
+            freq_spectrum_fft = channel.get_frequency_spectrum()
+            freq_spectrum_fft_copy = copy.copy(freq_spectrum_fft)  # copy spectrum so it is only changed within the trigger module
+            sampling_rate = channel.get_sampling_rate()
+
+            freq_spectrum_fft_copy *= f
+            trace_filtered = NuRadioReco.utilities.fft.freq2time(freq_spectrum_fft_copy, sampling_rate)
```

Comment:
There is a built-in functionality to return a bandpass-filtered copy of a channel - `channel.get_filtered_trace(passband, filter_type, order)` ([documentation](https://nu-radio.github.io/NuRadioMC/NuRadioReco/apidoc/NuRadioReco.framework.base_trace.html?highlight=get_filtered_trace#NuRadioReco.framework.base_trace.BaseTrace.get_filtered_trace)). Maybe it doesn't work with a non-default `rp` parameter for the chebyshev filter (yet), but otherwise I think using this function would be a bit tidier?

Reply:
Thank you! I didn't know that function. I added the ripple parameter rp. It really looks better now :)